modules/tactical/src/mutation.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Represents a mutation to be performed on top of an object.
+ */
+export interface Mutation {
+  
+  /**
+   * Applies the mutation to an object, returning success or failure.
+   */
+  apply(obj: any): boolean;
+}
+
+/**
+ * Sets a group of properties on an object, overwriting any previous values.
+ */
+export class SetPropertiesMutation implements Mutation {
+  changes: any;
+  
+  constructor(changes: any) {
+    this.changes = changes;
+  }
+  
+  apply(obj: any): boolean {
+    if (obj == null) {
+      return false;
+    }
+    var props = Object.getOwnPropertyNames(this.changes);
+    for (var i = 0; i < props.length; i++) {
+      var key: string = props[i];
+      obj[key] = this.changes[key];
+    }
+    return true;
+  }
+}
+
+/**
+ * Applies a set of mutations to the value of a child property on an object.
+ */
+export class SubPropertyMutation implements Mutation {
+  key: string;
+  mutations: Mutation[];
+  
+  constructor(key: string, mutations: Mutation[]) {
+    this.key = key;
+    this.mutations = mutations;
+  }
+  
+  apply(obj: any): boolean {
+    // Cannot apply mutations to something which isn't there.
+    if (!obj.hasOwnProperty(this.key)) {
```

Comment:
If `obj` can be anything, it might be a good idea to check if it has a `hasOwnProperty()` method (e.g. it could be null or undefined).